service/storage_service.cc

```diff
@@ -157,6 +157,15 @@ namespace {
 [[nodiscard]] locator::host_id_or_endpoint_list parse_node_list(const std::string_view comma_separated_list) {
     return string_list_to_endpoint_list(utils::split_comma_separated_list(comma_separated_list));
 }
+
+void check_raft_rpc_scheduling_group(auto& db, const std::string_view rpc_name) {
+    const auto gossip_scheduling_group = db.get_gossip_scheduling_group();
+    if (current_scheduling_group() != gossip_scheduling_group) {
+        slogger.warn("Raft group0 RPCs should be executed in the gossip scheduling group [{}], current group is [{}], operation [{}].",
+                gossip_scheduling_group.name(), current_scheduling_group().name(), current_scheduling_group().name(), rpc_name);
```

Comment:
Won't this result in massive spam in the logs during upgrades?
Maybe a better solution is to add a feature flag and guard the previous `on_internal_error()` with it.

Reply:
You can't. The way to handle upgrades is with feature flags: these are only set when all nodes have upgraded. Practically speaking, here you would only warn/assert if the feature flag was set: then you know the entire cluster is on the new version.